subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+// +build enterprise
+
+package serveraclinit
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+)
+
+// Test that ACL policies get updated if namespaces config changes.
+func TestRun_ACLPolicyUpdates(t *testing.T) {
```

Comment:
Also, it might be good to test the `createOrUpdateACLPolicy` method directly.

Reply:
> Also, it might be good to test the createOrUpdateACLPolicy method directly.

It looks like the only thing we're not testing is this:
```
		// This shouldn't happen, because we're looking for a policy
		// only after we've hit a `Policy already exists` error.
		if policy.ID == "" {
			return errors.New("Unable to find existing ACL policy")
		}
```

If that's true then not sure this warrants a separate test unless there's a case I'm missing.

Reply:
> It makes more sense to me to have the initial functionality tests before the update tests.

Down to move it but I'll do it after you have a second look so the changes are easier to review.

Reply:
I think those cases are already covered through the tests. I ran with code coverage:
![image](https://user-images.githubusercontent.com/1034429/74292111-6cd74580-4cf4-11ea-9711-59edae7e682a.png)

Reply:
This is an argument for doing comprehensive testing at the lowest possible level rather than up the call tree somewhere. Test organization rather than code coverage.

Reply:
Yes, I prefer testing the public interfaces and allowing the private implementation to vary. In my experience it makes the code easier to work with because you can refactor without having to update a ton of tests. It's a balance tradeoff in case your public interface tests get super complicated to exercise corner cases in helper methods. In my judgement it's not worth it in this case.